### PR TITLE
packages / tagbot / badges

### PR DESCRIPTION
**Pull request by JeffFessler**
especially for updated MIRTio that addresses a HDF5.jl issue

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -27,19 +27,19 @@ Wavelets = "29a6e085-ba6d-5f35-a997-948ac2efa89a"
 
 [compat]
 FFTViews = "0.3"
-FFTW = "1.2"
-FillArrays = "0.9, 0.10, 0.11"
+FFTW = "1.3"
```
Comment (johnnychen94) [inline Project.toml:30]:
```suggestion
FFTW = "1.2"
```

File: Project.toml [hunk 2]
```diff
@@ -27,19 +27,19 @@ Wavelets = "29a6e085-ba6d-5f35-a997-948ac2efa89a"
...
-MIRTio = "0.2"
+MIRTio = "0.3"
 MosaicViews = "0.2"
 NFFT = "0.6"
 Plots = "1.10"
 Reexport = "0.2, 1.0"
-SpecialFunctions = "0.10, 1"
-Wavelets = "0.9"
+SpecialFunctions = "1"
+Wavelets = "0.9.4"
```
Comment (johnnychen94) [inline Project.toml:42]:
```suggestion
Wavelets = "0.9"
```

**Review by johnnychen94 — COMMENTED** [review]
It's very possible that other (old) packages aren't compatible to FillArrays `0.11` and an over-restricted version compat can result in a very unpleasant experience during version resolving. For this reason, it's not advisable to drop old versions too eagerly.

Dropping old version often due to the following reasons:

* the feature we need only exists in the new version, and
* declaring compatibility to old version requires more maintenance work

According to https://semver.org/, MIRT should be compatible to `FFTW@1` and `Wavelets@0.9` since it doesn't require the new features there (does it?), and there's no extra maintenance burden here to declare the compatibility(it's the responsibility of FFTW and Wavelets to maintain the compatibility). Thus I made two suggestions to revert it. Hopefully, we could choose to not drop the old versions, but I'm not familiar with the MIRT internal, so you make the choices.

**Comment by JeffFessler** [comment]
@johnnychen94 thanks for your advice on this.  it passes the tests now (including older versions of the packages in compat) so can you please take one more look at this before i squash and merge?  i am having some git push issues so i probably messed up your authorship on this one so apologies for that.  my biggest question mark is whether the "Wavelets 0.9" is really the right entry given the updates we just did for Wavelets, but it seems to work so if you say it is ok then i am ready to proceed.

File: Project.toml [hunk 3]
```diff
@@ -27,13 +27,13 @@ Wavelets = "29a6e085-ba6d-5f35-a997-948ac2efa89a"
 
 [compat]
 FFTViews = "0.3"
-FFTW = "1.2"
+FFTW = "1.2, 1.3"
```
Comment (johnnychen94) [inline Project.toml:30]: `"1.2"` means  `[1.2.0, 2.0.0)`, which is a superset of `"1.3"`, i.e., `[1.3.0, 2.0.0)`.

I guess FFTW "1" also works?

```suggestion
FFTW = "1"
```

File: Manifest.toml [hunk 4]
```diff
@@ -0,0 +1,1138 @@
+# This file is machine-generated - editing it directly is not advised
```
Comment (johnnychen94) [inline Manifest.toml:1]: I think `Manifes.toml` is accidentally committed here, need to remove it before you merge this PR.

**Review by johnnychen94 — COMMENTED** [review]
MIRT has nothing to do with the version conflict for `SpecialFunctions`, so as long as Wavelets updates its compat section, Pkg can automatically figure out a happy version for it. Thus I would personally revert the entire `Project.toml` because it fixes nothing.

It all depends on how you expect this PR does, if you intend to change the compat entries, then it's okay.

File: Manifest.toml (hunk 4 above)
Comment (JeffFessler) [reply Manifest.toml]: i had added Manifest when I was pointing MIRT to my branch of Wavelets (I thought only the Manifest has the specific branch info, though I could be wrong about that).  but now that I am using the main branch of Wavelets again, I agree I don't need Manifest.  thanks.

**Pull request merged**